app/features/worksheet/tools.py

```diff
@@ -0,0 +1,278 @@
+                    logger.warning(f"Not unique {self.section_name}. Attempt {attempts} of {max_attempts}")
+                    continue
+                generated_questions.append(self.response)
+                if self.verbose:
+                    logger.info(f"Valid {self.section_name} added: {self.response}")
+                    logger.info(f"Total generated {self.section_name}(s): {len(generated_questions)}")
+            else:
+                if self.verbose:
+                    logger.warning(f"Invalid {self.section_name} response format. Attempt {attempts} of {max_attempts}")
+        
+        # Log if fewer questions are generated
+        if len(generated_questions) < num_questions:
+            logger.warning(f"Only generated {len(generated_questions)} out of {num_questions} requested {self.section_name}(s)")
+
+        return generated_questions
+
+# Multiple Choice Question subclass
+class MultipleChoiceQuestion(QuestionBase):
+    def set_names(self):
+        self.section_name = 'Miltiple-Choice question'
+        self.main_field = 'question'
+    
+    def sub_validate(self):
+        MultipleChoiceQuestionFormat(**self.response)
+        return True
+
+# Summary subclass
+class Summary(QuestionBase):
+    def set_names(self):
+        self.section_name = 'Summary'
+        self.main_field = 'description'
+    
+    def sub_validate(self):
+        SummaryFormat(**self.response)
+        return True
+
+# Fill in the Blank subclass
+class FillInTheBlankQuestion(QuestionBase):
+    def set_names(self):
+        self.section_name = 'Fill-in-the-blank question'
+        self.main_field = 'question'
+    
+    def sub_validate(self):
+        validated_response = FillinblankQuestionFormat(**self.response)
+        if '_' in getattr(validated_response, self.main_field):
+            return True
+        else:
+            if self.verbose:
+                logger.warning(f"{self.section_name} does not have blank ___.")
+            return False
+
+# Open ended question subclass
+class OpenEndedQuestion(QuestionBase):
+    def set_names(self):
+        self.section_name = 'Open-ended question'
+        self.main_field = 'question'
+    
+    def sub_validate(self):
+        OpenEndedQuestionFormat(**self.response)
+        return True
+
+# True False question subclass
+class TrueFalseQuestion(QuestionBase):
+    def set_names(self):
+        self.section_name = 'True or False question'
+        self.main_field = 'question'
+    
+    def sub_validate(self):
+        TrueFalseQuestionFormat(**self.response)
+        return True
+
+# Create worksheets
+class WorksheetBuilder:
+    def __init__(self, topic, grade_level, model=None, embedding_model = None, prompt_summary=None, parser_summary = None, prompt_fill_in_blank=None, parser_fill_in_blank=None, prompt_multiple_choice=None, parser_multiple_choice=None, prompt_open_ended = None, parser_open_ended = None, prompt_true_false = None, parser_true_false = None, verbose=False):
```

Comment:
Use a Pydantic Schema for decoupling args. and ensuring a better reusability and readability.